src/references.rs

```diff
@@ -125,46 +128,35 @@ fn check_nix_file(
 
     Ok(validation::sequence_(
         nix_file.syntax_root.syntax().descendants().map(|node| {
-            let text = node.text().to_string();
-            let line = nix_file.line_index.line(node.text_range().start().into());
-
             // We're only interested in Path expressions
-            let Some(path) = rnix::ast::Path::cast(node) else {
+            let Some(path) = rnix::ast::Path::cast(node.clone()) else {
```

Comment:
This clone makes me sad. I think it's only needed because of the to_problem closure. Could you tell me why it's there now?

Reply:
It was only needed because I moved `line` and `text` closer to where they were used without thinking much about how big the node object might be. I changed it back to remove the clone.

Reply:
# 🔥 
Love that.